### ISSUE TITLE
Unlisted buffers of regular files should change to listed upon loading in window.

### ISSUE THREAD

**Issue by strait** [neovim/neovim#12697]
- `nvim --version`: NVIM v0.5.0-590-gb39edb5b6
with hidden option on.

- Operating system/version: Linux

### Actual behaviour
I write a new file with ':w mynew.txt'. The new buffer created for this is unlisted'
I change to this buffer with ':b#'. The buffer is still unlisted.

### Expected behaviour
Loading the new buffer into a window should be enough to make this file listed without the extra ':edit' or ':set buflisted' commands.

**Comment by erw7** [comment]
You may need to be a little more creative, but does setting the `buflisted` option in `autocmd`, as shown below, not accomplish your goal?

```vim
autocmd BufWrite * :call setbufvar("#", '&buflisted', v:true)
autocmd BufWrite * :call setbufvar(bufnr(expand('<afile>')), '&buflisted', v:true)
```

**Comment by strait** [comment]
No, it seems your first autocmd sets the option on an unrelated buffer (the previous '#'), and your second sets the option on the new buffer, even if I never load it into a window.

**Comment by ahmadie** [comment]
autocmd BufEnter * :call setbufvar("%", '&buflisted', v:true)

what about top code ?

I do not think airline should miss up with buflisted array as there are plugins for tab management that manupilate buflisted and can cause conflicts, airline should only reflect what it can sees only and not do tab / buffer management in anyway.

**Comment by chrisbra** [comment]
What makes you think airline does?
